Add a package sdk-vanilla to the repository

lyleschemmerling opened the pull request:
Fixes #175

Add a new package `sdk-vanilla` to the repository to provide FusionAuth SDK for vanilla JavaScript with web components.

* **Package Metadata and Configuration**
  - Add `package.json` with package metadata, scripts, dependencies, and build configuration.
  - Add `vite.config.ts` for Vite build configuration.

* **Web Components**
  - Add `fa-account.ts` to implement the `fa-account` web component with a button to redirect to the user's account management page.
  - Add `fa-login.ts` to implement the `fa-login` web component with a button to redirect to the /app/login endpoint and start the OAuth flow.
  - Add `fa-logout.ts` to implement the `fa-logout` web component with a button to redirect to the /app/logout endpoint.
  - Add `fa-register.ts` to implement the `fa-register` web component with a button to redirect to the /app/register endpoint.

* **FusionAuthService**
  - Add `FusionAuthService.ts` to implement the `FusionAuthService` class with methods for login, register, logout, manage account, fetch user info, refresh token, auto-refresh, and post-redirect handling.
  - Store and retrieve configuration from localStorage.

* **Documentation**
  - Add `README.md` with instructions and examples on how to use the `sdk-vanilla` package.

* **Tests**
  - Add tests for `FusionAuthService` in `FusionAuthService.test.ts`.
  - Add tests for web components in `fa-account.test.ts`, `fa-login.test.ts`, `fa-logout.test.ts`, and `fa-register.test.ts`.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/FusionAuth/fusionauth-javascript-sdk/pull/176?shareId=f0b26265-9b2b-46c7-98e1-3814909bfd72).

lyleschemmerling committed:
Add a package sdk-vanilla to the repository

Fixes #175

Add a new package `sdk-vanilla` to the repository to provide FusionAuth SDK for vanilla JavaScript with web components.

* **Package Metadata and Configuration**
  - Add `package.json` with package metadata, scripts, dependencies, and build configuration.
  - Add `vite.config.ts` for Vite build configuration.

* **Web Components**
  - Add `fa-account.ts` to implement the `fa-account` web component with a button to redirect to the user's account management page.
  - Add `fa-login.ts` to implement the `fa-login` web component with a button to redirect to the /app/login endpoint and start the OAuth flow.
  - Add `fa-logout.ts` to implement the `fa-logout` web component with a button to redirect to the /app/logout endpoint.
  - Add `fa-register.ts` to implement the `fa-register` web component with a button to redirect to the /app/register endpoint.

* **FusionAuthService**
  - Add `FusionAuthService.ts` to implement the `FusionAuthService` class with methods for login, register, logout, manage account, fetch user info, refresh token, auto-refresh, and post-redirect handling.
  - Store and retrieve configuration from localStorage.

* **Documentation**
  - Add `README.md` with instructions and examples on how to use the `sdk-vanilla` package.

* **Tests**
  - Add tests for `FusionAuthService` in `FusionAuthService.test.ts`.
  - Add tests for web components in `fa-account.test.ts`, `fa-login.test.ts`, `fa-logout.test.ts`, and `fa-register.test.ts`.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/FusionAuth/fusionauth-javascript-sdk/issues/175?shareId=XXXX-XXXX-XXXX-XXXX).

lyleschemmerling commented on packages/sdk-vanilla/README.md:
```diff
@@ -0,0 +1,89 @@
+  </script>
+</head>
+<body>
+  <fa-login></fa-login>
```
please add all component buttons to this example

mmanes replied:
Please add all component buttons to this example

lyleschemmerling committed:
Add a section to the README.md file to address the comments on the pull request

* **New `sdk-vanilla` package**
  - Add the package to the repository
  - Include specified web components (`fa-account`, `fa-login`, `fa-logout`, `fa-register`) and a `FusionAuthService` typescript class
  - Implement the `FusionAuthService` class to handle intermediate functions and use the same functions from the core package as the other SDKs
  - Match the styles of the web components with the existing SDKs
  - Add a `vite.config.ts` file to build the library
  - Create documentation and a `README.md` file with instructions and examples on how to use the `sdk-vanilla` package
  - Update the `FusionAuthService` class to store the config in localStorage and pull the config from localStorage before making any requests
  - Ensure each of the "start" methods in the service uses `getConfig` before making requests
  - Remove the need for components to pass a config into the constructor for the service
  - Add tests for the `FusionAuthService` class and the web components

lyleschemmerling commented:
Please add build scripts for yalc in the top-level package.json for sdk-vanilla

lyleschemmerling committed:
Add section to README.md to address new comments and add build scripts for sdk-vanilla

* **README.md**
  - Add a section to address new comments
  - Include details about the new `sdk-vanilla` package, documentation updates, and additional examples
  - Mention performance optimizations and styling updates for web components
  - Update Vite configuration and tests for `FusionAuthService` and web components

* **package.json**
  - Add `sdk-vanilla` to workspaces
  - Add build script for `sdk-vanilla`
  - Add yalc publish script for `sdk-vanilla`
  - Update test script to include `sdk-vanilla`
